File: couch/ruby/read/couch_read.rb
#! /usr/bin/ruby
# -*- coding: utf-8 -*-
#
#	couch_read.rb
#
#					Feb/24/2015
#
require 'restclient'
#
require 'json'
#
load '/var/www/data_base/common/ruby_common/text_manipulate.rb'
# ---------------------------------------------------------------------
#
puts	"*** 開始 ***"
#
url_collection = 'http://localhost:5984/nagano'
url_docs  =  url_collection + '/_all_docs'
puts url_docs
str_json = RestClient.get url_docs
#
list_aa=JSON.parse(str_json)
#
dict_aa={}
list_aa['rows'].each {|row|
	key = row['key']
	url_target  =  url_collection + '/' + key
	str_json = RestClient.get url_target
	unit_aa=JSON.parse(str_json)
	dict_aa[key]=unit_aa
	}
#
dict_display_proc (dict_aa)
puts	"*** 終了 ***"
#

File: couch/python/read/couch_read.py
#! /usr/bin/python3
# -*- coding: utf-8 -*-
#
#	read/couch_read.py
#
#					Aug/01/2017
#
# ----------------------------------------------------------------
import os
import sys
#
sys.path.append('../../../common/python_common')
from text_manipulate import dict_display_proc
#
from couch_manipulate import couch_to_dict_proc
# ----------------------------------------------------------------
sys.stderr.write("*** 開始 ***\n")
#
os.environ["http_proxy"]=''
#
url_json = 'http://localhost:5984/nagano'
#
dict_aa = couch_to_dict_proc(url_json)
#
dict_display_proc(dict_aa)
#
sys.stderr.write("*** 終了 ***\n")
# ----------------------------------------------------------------
